File: 05-use-cases/schedule-intelligence/scripts/test-schedule-extraction.js
#!/usr/bin/env node

/**
 * 일정 추출 테스트 스크립트 (Node.js)
 *
 * 실행 방법: node test-schedule-extraction.js
 */

const fs = require('fs');
const path = require('path');

// 색상 코드
const colors = {
  blue: '\x1b[34m',
  green: '\x1b[32m',
  yellow: '\x1b[33m',
  red: '\x1b[31m',
  reset: '\x1b[0m'
};

console.log(`${colors.blue}========================================${colors.reset}`);
console.log(`${colors.blue}일정 추출 테스트 스크립트 (Node.js)${colors.reset}`);
console.log(`${colors.blue}========================================\n${colors.reset}`);

// n8n webhook URL (사용자가 수정해야 함)
const WEBHOOK_URL = 'https://your-n8n-instance.com/webhook/schedule-intelligence';

// 샘플 이메일 디렉토리
const SAMPLE_DIR = path.join(__dirname, '..', 'data', 'sample-emails');

// 이메일 파일 목록
const EMAIL_FILES = [
  'meeting-simple.txt',
  'meeting-vague.txt',
  'deadline.txt',
  'event.txt'
];

/**
 * Webhook 호출 함수
 */
async function sendToWebhook(payload) {
  // 실제 환경에서는 주석 해제
  // const response = await fetch(WEBHOOK_URL, {
  //   method: 'POST',
  //   headers: { 'Content-Type': 'application/json' },
  //   body: JSON.stringify(payload)
  // });
  // return await response.json();

  // 테스트 모드
  return { success: true, message: 'Test mode - webhook not called' };
}

/**
 * 이메일 처리 함수
 */
async function processEmail(emailFile) {
  const filePath = path.join(SAMPLE_DIR, emailFile);

  if (!fs.existsSync(filePath)) {
    console.log(`${colors.red}❌ 파일을 찾을 수 없습니다: ${filePath}${colors.reset}`);
    return;
  }

  console.log(`${colors.green}처리 중: ${emailFile}${colors.reset}`);

  // 이메일 내용 읽기
  const emailContent = fs.readFileSync(filePath, 'utf8');

  // JSON 페이로드 생성
  const payload = {
    email_content: emailContent,
    source_file: emailFile,
    received_at: new Date().toISOString()
  };

  // Webhook 호출
  const response = await sendToWebhook(payload);

  // 결과 출력
  console.log(`${colors.blue}📤 Webhook 페이로드:${colors.reset}`);
  console.log(JSON.stringify(payload, null, 2));

  console.log(`${colors.green}✅ 처리 완료\n${colors.reset}`);
  console.log('---');
}

/**
 * 메인 실행 함수
 */
async function main() {
  console.log(`${colors.yellow}📧 샘플 이메일 처리 시작...\n${colors.reset}`);

  for (const emailFile of EMAIL_FILES) {
    await processEmail(emailFile);
    // 다음 이메일 처리 전 대기
    await new Promise(resolve => setTimeout(resolve, 1000));
  }

  console.log(`\n${colors.green}========================================${colors.reset}`);
  console.log(`${colors.green}모든 테스트 완료!${colors.reset}`);
  console.log(`${colors.green}========================================\n${colors.reset}`);

  console.log(`${colors.yellow}💡 실제 테스트를 위해:${colors.reset}`);
  console.log('1. WEBHOOK_URL을 실제 n8n webhook 주소로 변경하세요');
  console.log('2. sendToWebhook 함수의 fetch 호출 주석을 해제하세요 (line 41-46)');
  console.log('');
}

// 실행
main().catch(console.error);

File: 05-use-cases/schedule-intelligence/scripts/test-schedule-extraction.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

"""
일정 추출 테스트 스크립트 (Python)

실행 방법: python test-schedule-extraction.py
"""

import json
import os
import time
from datetime import datetime
from pathlib import Path

# 색상 코드
class Colors:
    BLUE = '\033[34m'
    GREEN = '\033[32m'
    YELLOW = '\033[33m'
    RED = '\033[31m'
    RESET = '\033[0m'

def print_header():
    """헤더 출력"""
    print(f"{Colors.BLUE}========================================{Colors.RESET}")
    print(f"{Colors.BLUE}일정 추출 테스트 스크립트 (Python){Colors.RESET}")
    print(f"{Colors.BLUE}========================================\n{Colors.RESET}")

def send_to_webhook(payload):
    """
    Webhook 호출 함수

    실제 환경에서는 requests 라이브러리 사용:
    import requests
    response = requests.post(WEBHOOK_URL, json=payload)
    return response.json()
    """
    # 테스트 모드
    return {"success": True, "message": "Test mode - webhook not called"}

def process_email(email_file, sample_dir):
    """이메일 처리 함수"""
    file_path = os.path.join(sample_dir, email_file)

    if not os.path.exists(file_path):
        print(f"{Colors.RED}❌ 파일을 찾을 수 없습니다: {file_path}{Colors.RESET}")
        return

    print(f"{Colors.GREEN}처리 중: {email_file}{Colors.RESET}")

    # 이메일 내용 읽기
    with open(file_path, 'r', encoding='utf-8') as f:
        email_content = f.read()

    # JSON 페이로드 생성
    payload = {
        "email_content": email_content,
        "source_file": email_file,
        "received_at": datetime.now().isoformat()
    }

    # Webhook 호출
    response = send_to_webhook(payload)

    # 결과 출력
    print(f"{Colors.BLUE}📤 Webhook 페이로드:{Colors.RESET}")
    print(json.dumps(payload, ensure_ascii=False, indent=2))

    print(f"{Colors.GREEN}✅ 처리 완료\n{Colors.RESET}")
    print("---")

def main():
    """메인 실행 함수"""
    print_header()

    # n8n webhook URL (사용자가 수정해야 함)
    WEBHOOK_URL = "https://your-n8n-instance.com/webhook/schedule-intelligence"

    # 샘플 이메일 디렉토리
    script_dir = Path(__file__).parent
    sample_dir = script_dir.parent / 'data' / 'sample-emails'

    # 이메일 파일 목록
    email_files = [
        'meeting-simple.txt',
        'meeting-vague.txt',
        'deadline.txt',
        'event.txt'
    ]

    print(f"{Colors.YELLOW}📧 샘플 이메일 처리 시작...\n{Colors.RESET}")

    for email_file in email_files:
        process_email(email_file, sample_dir)
        # 다음 이메일 처리 전 대기
        time.sleep(1)

    print(f"\n{Colors.GREEN}========================================{Colors.RESET}")
    print(f"{Colors.GREEN}모든 테스트 완료!{Colors.RESET}")
    print(f"{Colors.GREEN}========================================\n{Colors.RESET}")

    print(f"{Colors.YELLOW}💡 실제 테스트를 위해:{Colors.RESET}")
    print("1. WEBHOOK_URL을 실제 n8n webhook 주소로 변경하세요")
    print("2. send_to_webhook 함수에서 requests 라이브러리 사용 (line 36-39)")
    print("3. requests 설치: pip install requests")
    print("")

if __name__ == "__main__":
    main()
